presentation/src/main/java/com/architect/coders/mu8/di/ApplicationComponent.kt

```diff
@@ -13,7 +13,7 @@ import javax.inject.Singleton
 @Component(modules = [DomainModule::class, DataModule::class])
 interface ApplicationComponent {
 
-    fun plus(module: CharactersViewModelModule): CharactersViewModelComponent
+    fun plus(module: ViewModelModule): ViewModelComponent
```

Comment:
@recaldev  I didn't understand it, could you explain us  the `@Component.factory` and the plus function

Reply:
@jamacado can you explain this part?

Reply:
@aliceresponde @recaldev he change the name to make it more generic and in ViewModelComponent there are all the viewmodels to be injected. @BindsInstance creates the instance only when needed